File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/composite/FormTests.java
package com.epam.jdi.uitests.testing.unittests.tests.composite;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.pageobjects.pages.IContactForm;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.Contact.DEFAULT;
import static com.epam.jdi.uitests.testing.unittests.enums.Buttons.SUBMIT;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.contactFormPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.checkResult;
import static com.epam.web.matcher.testng.Assert.*;
import static java.lang.String.format;

/**
 * Created by Dmitry_Lebedev1 on 10/15/2015.
 */
public class FormTests extends InitTests {
    Supplier<IContactForm> form = () -> contactFormPage.contactForm;

    public FormTests() { }
    public FormTests(Supplier<IContactForm> form) {
        this.form = form;
    }

    @BeforeMethod
    public void before(final Method method) {
        isInState(CONTACT_PAGE, method);
    }

    @Test
    public void fillTest() {
        form.get().fill(DEFAULT);
        assertEquals(form.get().getFormValue(), DEFAULT.toList());
    }

    @Test
    public void submitTest() {
        form.get().submit(DEFAULT);
        checkResult(DEFAULT.toString());
    }

    @Test
    public void submitSpecButtonStringTest() {
        form.get().submit(DEFAULT, "submit");
        checkResult(DEFAULT.toString());
    }

    @Test
    public void submitSpecButtonEnumTest() throws Exception {
        form.get().submit(DEFAULT, SUBMIT);
        checkResult(DEFAULT.toString());
    }

    @Test
    public void submitStringTest() throws Exception {
        form.get().submit(DEFAULT.name);
        checkResult(format("Summary: 3\nName: %s", DEFAULT.name));
    }

    @Test
    public void verifyTest() {
        form.get().fillForm(DEFAULT);
        isEmpty(form.get().verify(DEFAULT));
    }

    @Test
    public void checkTest() {
        form.get().fillForm(DEFAULT);
        hasNoExceptions(() -> form.get().check(DEFAULT));
    }
}

File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/Tests/Composite/FormTests.cs
using NUnit.Framework;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_UIWebTests.UIObjects.TestSite;
using JDI_UIWebTests.Entities;
using JDIWebTests.UIObjects.Sections;
using System.Collections.Generic;
using static JDI_UIWebTests.Tests.Complex.CommonActionsData;
using static JDI_UIWebTests.Enums.Buttons;
using JDI_Matchers.NUnit;

namespace JDI_UIWebTests.Tests.Composite
{
    public class FormTests
    {
        private ContactForm _contactForm = ContactFormPage.ContactForm;

        [SetUp]
        public void SetUp()
        {
            Logger.Info("Navigating to Contact page.");
            ContactFormPage.Open();
            ContactFormPage.CheckTitle();
            ContactFormPage.IsOpened();
            Logger.Info("Setup method finished");
            Logger.Info("Start test: " + TestContext.CurrentContext.Test.Name);
        }

        [Test]
        public void FillFormTest()
        {
            _contactForm.Fill(Contact.DEFAULT_CONTACT);            
            IList<string> filledFilds = _contactForm.GetFormValue();            
            new Check().CollectionEquals(filledFilds, Contact.DEFAULT_CONTACT.ToList());            
        }

        [Test]
        public void SubmitTest()
        {
            _contactForm.Submit(Contact.DEFAULT_CONTACT);            
            CheckResult(Contact.DEFAULT_CONTACT.ToString());
        }

        [Test]
        public void SubmitSpecButtonStringTest()
        {
            _contactForm.Submit(Contact.DEFAULT_CONTACT, "submit");
            CheckResult(Contact.DEFAULT_CONTACT.ToString());
        }

        [Test]
        public void SubmitSpecButtonEnumTest()
        {
            _contactForm.Submit(Contact.DEFAULT_CONTACT, SUBMIT);
            CheckResult(Contact.DEFAULT_CONTACT.ToString());
        }

        [Test]
        public void SubmitStringTest()
        {
            _contactForm.Submit(Contact.DEFAULT_CONTACT.FirstName);
            string s = string.Format("Summary: 3\r\nName: {0}",
                         Contact.DEFAULT_CONTACT.FirstName);
            CheckResult(s);
        }

        [Test]
        public void VerifyTest()
        {
            _contactForm.Fill(Contact.DEFAULT_CONTACT);            
            new Check().IsTrue(_contactForm.Verify(Contact.DEFAULT_CONTACT).Count == 0);            
        }

 
        [Test]
        public void checkTest()
        {
            _contactForm.Fill(Contact.DEFAULT_CONTACT);
            new Check().HasNoException(() => _contactForm.Check(Contact.DEFAULT_CONTACT));            
        }
 

    }
}
